### PR TITLE
Deprecate Conv2d, ConvTranspose2d and BatchNorm2d

### PR DESCRIPTION
**Pull request by fmassa**
PyTorch now natively supports empty batches for `Conv2d`, `ConvTranspose2d` and `BatchNorm2d`, so no need to have our own versions anymore.

Follow-up from #2215 which removed those ops. I'm adding the functions back to avoid breaking BC, but I'm raising a error instead to encourage users to replace those ops in their code.

### CODE REVIEW

**Comment by fmassa** [comment]
@pmeier I've started seeing `python_type_check` errors recently, do you have an idea of what this might be?

**Comment by pmeier** [comment]
> @pmeier I've started seeing `python_type_check` errors recently, do you have an idea of what this might be?

I'll investigate and get back to you.

**Comment by fmassa** [comment]
@pmeier we should note that the CI job for `python_type_check` doesn't compile nor install torchvision, so `setup.py` is not invoked, so this error is kind of expected I think, following your explanation.

What puzzles me is that this CI was working before and just recently stopped working, and I don't have any ideas why that would be the case.

**Comment by pmeier** [comment]
> CI job for `python_type_check` doesn't compile nor install torchvision, so `setup.py` is not invoked,

Isn't `L134` below doing exactly that?

https://github.com/pytorch/vision/blob/cb65590091ac357ef1ff28b8f97a132683b60c37/.circleci/config.yml.in#L131-L135

**Comment by fmassa** [comment]
Oh, you are right, I missed that line. Let me have a closer look at the commit history

**Comment by fmassa** [comment]
Let's see if https://github.com/pytorch/vision/pull/2246 fixes it (I'm not sure)
